libbeat/processors/add_cloud_metadata/provider_aws_ec2.go

```diff
@@ -97,10 +92,7 @@ func fetchRawProviderMetadata(
 	awsRegion := instanceIdentity.InstanceIdentityDocument.Region
 	awsConfig.Region = awsRegion
 
-	clusterName, err := fetchEC2ClusterNameTag(awsConfig, instanceIdentity.InstanceIdentityDocument.InstanceID)
-	if err != nil {
-		logger.Warnf("error fetching cluster name metadata: %s.", err)
-	}
+	clusterName, _ := fetchEC2ClusterNameTag(awsConfig, instanceIdentity.InstanceIdentityDocument.InstanceID)
```

Comment:
We probably should not ignore the error here. When we do collect tags from an ec2 instance, this error should be surfaced. For example user might not have DescribeTags permission configured properly.

Reply:
We should keep this warning here. Also for the original log, since we removed the `logger.Warnf("error fetching EC2 Identity Document: %s.", err)` line and it has a `return` there at line 88, this warning should not matter when EC2 is not the proper cloud metadata source.

Reply:
You're right @kaiyan-sheng , I added the log message again to this one